### PR TITLE
Fix esptool imports (RDT-222)

### PR DESCRIPTION
**Pull request by hmalpani**
This MR fixes esptool imports.

https://github.com/espressif/pytest-embedded/pull/77#discussion_r834197044

### CODE REVIEW

**Comment by hmalpani** [comment]
@hfudev PTAL

File: pytest-embedded-idf/pytest_embedded_idf/app.py [hunk 1]
```diff
@@ -214,7 +214,8 @@ def get_sha256(self, filepath: str) -> Optional[str]:
         Returns:
             sha256 appended to app
         """
-        from esptool import LoadFirmwareImage, hexify
+        from esptool.bin_image import LoadFirmwareImage
+        from esptool.util import hexify
```
Comment (hfudev) [inline pytest-embedded-idf/pytest_embedded_idf/app.py:218]: could you use the `self.ESPTOOL_VERSION == EsptoolVersion.V3` or `self.ESPTOOL_VERSION == EsptoolVersion.V4` to determine the correct import here? We still need to make the code compatible with esptool 3.x.

**Review by hfudev — CHANGES_REQUESTED** [review]

File: pytest-embedded-idf/pytest_embedded_idf/app.py (hunk 1 above)
Comment (hmalpani) [reply pytest-embedded-idf/pytest_embedded_idf/app.py]: Updated the PR

File: pytest-embedded-idf/pytest_embedded_idf/app.py [hunk 2]
```diff
@@ -214,7 +214,15 @@ def get_sha256(self, filepath: str) -> Optional[str]:
         Returns:
             sha256 appended to app
         """
-        from esptool import LoadFirmwareImage, hexify
+        from pytest_embedded_serial_esp.serial import EsptoolVersion
+
+        from .serial import IdfSerial
```
Comment (hfudev) [inline pytest-embedded-idf/pytest_embedded_idf/app.py:219]: sorry to be picky...

```python
from pytest_embedded_serial_esp.serial import EspSerial, EsptoolVersion

if EspSerial.ESPTOOL_VERSION == ...
```

would be simpler.

**Review by hfudev — CHANGES_REQUESTED** [review]

**Pull request merged**